### ISSUE TITLE
Boolean Mutation incorrectly survives

### ISSUE THREAD

**Issue by robertlarkins** [stryker-mutator/stryker-net#842]
**Describe the bug**
When running Stryker against the following code:

```csharp
public static bool operator ==(ValueObject? left, ValueObject? right)
{
    var isLeftNull = left is null;
    var isRightNull = right is null;

    if (isLeftNull && isRightNull)
    {
        return true;
    }

    if (isLeftNull || isRightNull)
    {
        return false;
    }

    return left!.Equals(right);
}
```
Stryker incorrectly reports that the boolean mutation from `true` to `false` in 
```csharp
if (isLeftNull && isRightNull)
{
    return true;
}
```
survives. If I manually change this to `return false;` I have a test that does fail.

**Logs**
[log-20191209.txt](https://github.com/stryker-mutator/stryker-net/files/3937675/log-20191209.txt)

**Expected behavior**
This Boolean mutation is reported as killed.

**Desktop (please complete the following information):**
 - OS: Windows
 - Core 3.1 Library project and xUnit test project, both with Nullable enabled.
 - Stryker Version: beta 0.14.3

**Additional context**
If useful a solution has been created that shows this bug here: https://github.com/robertlarkins/code-showing-stryker-bug

**Comment by dupdob** [comment]
Thanks for reporting this. And thanks for providing a project to reproduce it. 
I confirm I am able reproduce it. As I suspected reading your code, this is an issue with coverage analysis, as turning it off ('-ca off' option on the command line) restore the expected behaviour.

I suggest you use this option while we are working on a fix. Testing will increase, but results will be reliable.

**Comment by Mobrockers** [comment]
Since the issue has been identified and a workaround has been supplied i'll go ahead and close this issue. The problem will be further investigated and fixed if possible in #843
